### PR TITLE
Add Kit UI to KitPVP

### PR DESCRIPTION
**Pull request by Roblox-Thot**
<div align="center">

</div>

### CODE REVIEW

**Comment by DataM0del** [comment]
Completely un-relevant PR description. Would this be possible to do with the NPCs?

File: index.js [hunk 1]
```diff
@@ -91,7 +92,7 @@ function updateChunks(x, z, client) {
 		for (let checkZ = -7; checkZ < 7; checkZ++) {
 			const pos = [Math.floor(x / 16) + checkX, Math.floor(z / 16) + checkZ];
 			currentlyLoaded.push(pos.join());
-			if (chunks.includes(pos.join()) || queuedChunks.includes(pos.join())) continue;
+			if (chunks.includes(pos.join()) || queuedChunks.includes(pos.join())) {continue;}
```
Comment (DataM0del) [inline index.js:95]:
```suggestion
			if (chunks.includes(pos.join()) || queuedChunks.includes(pos.join())) continue;
```

File: index.js [hunk 2]
```diff
@@ -75,8 +76,8 @@ const VERSION = "3.35.38", CELL_VOLUME = 16 * 16 * 16;
 const DEG2RAD = Math.PI / 180, RAD2DEG = 180 / Math.PI;
 
 function translateText(text) {
-	for (const [code, color] of Object.entries(COLOR_CODES)) text = text.replaceAll(code, color);
-	return text;
+	for (const [code, color] of Object.entries(COLOR_CODES)) new_text = text.replaceAll(code, color);
+	return new_text;
```
Comment (DataM0del) [inline index.js:80]:
```suggestion
	for (const [code, color] of Object.entries(COLOR_CODES)) text = text.replaceAll(code, color);
	return text;
```
why are you changing random stuff lol

File: index.js [hunk 3]
```diff
@@ -136,7 +137,7 @@ function createChunk(packet) {
 	const chunk = new Chunk();
 	for (const cell of packet.cells) {
 		const array = new BitArray(CELL_VOLUME, cell.bitsPerEntry, cell.bitArray);
-		if (!array) continue;
+		if (!array) {continue;}
```
Comment (DataM0del) [inline index.js:140]:
```suggestion
		if (!array) continue;
```

File: index.js [hunk 4]
```diff
@@ -145,7 +146,7 @@ function createChunk(packet) {
 					for (let skyY = 0; skyY < 256; skyY++) {
 						chunk.setSkyLight(new Vec3(x, skyY, z), 15);
 					}
-					if (cell.palette.length <= 0) continue;
+					if (cell.palette.length <= 0) {continue;}
```
Comment (DataM0del) [inline index.js:149]:
```suggestion
					if (cell.palette.length <= 0) continue;
```

File: index.js [hunk 5]
```diff
@@ -194,24 +195,24 @@ function purgePlayers(client) {
 }
 
 function convertToByte(num) {
-	num = num & 0xFF;
-	num = num > 127 ? num - 256 : num;
-	return num;
+	num &= 0xFF;
+	new_num = num > 127 ? num - 256 : num;
+	return new_num;
```
Comment (DataM0del) [inline index.js:200]:
```suggestion
	num = num & 0xFF;
	num = num > 127 ? num - 256 : num;
	return num;
```
don't need unrelated changes, that should be in a different PR.

File: index.js [hunk 6]
```diff
@@ -194,24 +195,24 @@ function purgePlayers(client) {
...
 }
 
 function convertAngle(ang, num) {
-	ang = ang / 256 * Math.PI * 2;
-	ang = (((ang * -1) * RAD2DEG) - (num != undefined ? num : 0)) * 256 / 360;
-	return convertToByte(ang);
+	new_ang = ang / 256 * Math.PI * 2;
+	new_ang = (((ang * -1) * RAD2DEG) - (num != undefined ? num : 0)) * 256 / 360;
+	return convertToByte(new_ang);
```
Comment (DataM0del) [inline index.js:206]:
```suggestion
	ang = ang / 256 * Math.PI * 2;
	ang = (((ang * -1) * RAD2DEG) - (num != undefined ? num : 0)) * 256 / 360;
	return convertToByte(ang);
```
reverting unrelated changes not related to the PR v999

File: index.js [hunk 7]
```diff
@@ -194,24 +195,24 @@ function purgePlayers(client) {
...
 function spawnEntity(packet, client) {
 	playerRotations[packet.id] = [convertAngle(packet.yaw, 180), convertAngle(packet.pitch)];
 	playerGamemodes[packet.id] = GAMEMODES[packet.gamemode ?? "survival"];
 	playerSkins[packet.id] = packet.cosmetics.skin;
-	if (playerUUIDs[packet.id] == undefined || playerPositions[packet.id] == undefined) return;
-	if (playerGamemodes[packet.id] == GAMEMODES.spectator) return;
-	if (players.includes(packet.id)) return;
+	if (playerUUIDs[packet.id] == undefined || playerPositions[packet.id] == undefined) {return;}
+	if (playerGamemodes[packet.id] == GAMEMODES.spectator) {return;}
+	if (players.includes(packet.id)) {return;}
```
Comment (DataM0del) [inline index.js:215]:
```suggestion
	if (playerUUIDs[packet.id] == undefined || playerPositions[packet.id] == undefined) return;
	if (playerGamemodes[packet.id] == GAMEMODES.spectator) return;
	if (players.includes(packet.id)) return;
```
NO MORE BARAKEETS!!!

File: index.js [hunk 8]
```diff
@@ -228,7 +229,7 @@ function spawnEntity(packet, client) {
 
 function sendActions() {
 	const newStates = {punching: lStates[0] > Date.now(), sprinting: lStates[1], sneak: lStates[2]};
-	if (newStates.punching == lastLState.punching && newStates.sprinting == lastLState.sprinting && newStates.sneak == lastLState.sneak) return;
+	if (newStates.punching == lastLState.punching && newStates.sprinting == lastLState.sprinting && newStates.sneak == lastLState.sneak) {return;}
```
Comment (DataM0del) [inline index.js:232]:
```suggestion
	if (newStates.punching == lastLState.punching && newStates.sprinting == lastLState.sprinting && newStates.sneak == lastLState.sneak) return;
```
Bart, say the line again...
> NO MORE BRACKETS!

File: index.js [hunk 9]
```diff
@@ -260,7 +261,7 @@ function disconnect() {
 async function connect(client, requeue, gamemode) {
 	if (requeue) {
 		skipKick = Date.now() + 20;
-		if (ClientSocket.socket) ClientSocket.disconnect();
+		if (ClientSocket.socket) {ClientSocket.disconnect();}
```
Comment (DataM0del) [inline index.js:264]:
```suggestion
		if (ClientSocket.socket) ClientSocket.disconnect();
```
's/}/' (real)

File: index.js [hunk 10]
```diff
@@ -319,7 +320,7 @@ async function connect(client, requeue, gamemode) {
 		if (!packet.canConnect) {
 			client.end(packet.errorMessage ?? "Disconnected");
 		}
-		if (requeue) return;
+		if (requeue) {return;}
```
Comment (DataM0del) [inline index.js:323]:
```suggestion
		if (requeue) return;
```
gahhh this must have been a formatter, right?

File: index.js [hunk 11]
```diff
@@ -342,7 +343,7 @@ async function connect(client, requeue, gamemode) {
 
 	// MINIBLOX SERVER
 	ClientSocket.on("disconnect", reason => {
-		if (skipKick > Date.now()) return;
+		if (skipKick > Date.now()) {return;}
```
Comment (DataM0del) [inline index.js:346]:
```suggestion
		if (skipKick > Date.now()) return;
```

File: index.js [hunk 12]
```diff
@@ -388,7 +389,11 @@ async function connect(client, requeue, gamemode) {
 		});
 	})
 	ClientSocket.on("CPacketMessage", packet => {
-		if (packet.text) client.write('chat', {message: JSON.stringify({text: translateText(packet.text)})});
+		if (packet.text) {
+			client.write('chat', {
+				message: JSON.stringify({text: translateText(packet.text)})
+			});
+		}
```
Comment (DataM0del) [inline index.js:396]:
```suggestion
		if (packet.text) client.write('chat', {message: JSON.stringify({text: translateText(packet.text)})});
```
revert the formatter v5

File: index.js [hunk 13]
```diff
@@ -494,8 +499,8 @@ async function connect(client, requeue, gamemode) {
 		}
 	});
 	ClientSocket.on("CPacketSpawnEntity", packet => {
-		if (ENTITIES[packet.type] == undefined) return;
-		if (packet.motion == undefined) packet.motion = {x: 0, y: 0, z: 0};
+		if (ENTITIES[packet.type] == undefined) {return;}
+		if (packet.motion == undefined) {packet.motion = {x: 0, y: 0, z: 0};}
```
Comment (DataM0del) [inline index.js:503]:
```suggestion
		if (ENTITIES[packet.type] == undefined) return;
		if (packet.motion == undefined) packet.motion = {x: 0, y: 0, z: 0};
```
more barakeet removing

File: index.js [hunk 14]
```diff
@@ -669,7 +674,7 @@ async function connect(client, requeue, gamemode) {
 		});
 	});
 	ClientSocket.on("CPacketBlockUpdate", packet => {
-		if (!chunks.includes([Math.floor(packet.x / 16), Math.floor(packet.z / 16)].join())) return;
+		if (!chunks.includes([Math.floor(packet.x / 16), Math.floor(packet.z / 16)].join())) {return;}
```
Comment (DataM0del) [inline index.js:677]:
```suggestion
		if (!chunks.includes([Math.floor(packet.x / 16), Math.floor(packet.z / 16)].join())) return;
```

File: index.js [hunk 15]
```diff
@@ -718,7 +723,7 @@ async function connect(client, requeue, gamemode) {
 	});
 	ClientSocket.on("CPacketEntityEquipment", packet => {
 		for (const equip of packet.equipment) {
-			if (equip.slot == 2) continue;
+			if (equip.slot == 2) {continue;}
```
Comment (DataM0del) [inline index.js:726]:
```suggestion
			if (equip.slot == 2) continue;
```

File: index.js [hunk 16]
```diff
@@ -834,7 +839,7 @@ async function connect(client, requeue, gamemode) {
 		}
 	});
 	ClientSocket.on("CPacketUpdateScoreboard", packet => {
-		if (scoreData[packet.index] == undefined) return;
+		if (scoreData[packet.index] == undefined) {return;}
```
Comment (DataM0del) [inline index.js:842]:
```suggestion
		if (scoreData[packet.index] == undefined) return;
```

File: index.js [hunk 17]
```diff
@@ -868,7 +894,7 @@ server.on('playerJoin', async function(client) {
 	}
 
 	client.on("end", function() {
-		if (ClientSocket.socket) ClientSocket.disconnect();
+		if (ClientSocket.socket) {ClientSocket.disconnect();}
```
Comment (DataM0del) [inline index.js:897]:
```suggestion
		if (ClientSocket.socket) ClientSocket.disconnect();
```

File: index.js [hunk 18]
```diff
@@ -988,9 +1014,18 @@ server.on('playerJoin', async function(client) {
...
+				ClientSocket.sendPacket(new SPacketMessage({text: "/kit "+ itemName.toLocaleLowerCase()}));
+				ClientSocket.sendPacket(new SPacketCloseWindow({windowId: 1}))
+				client.write('close_window', {windowId: 255})
+			}
+		}
 		let slot = Number.parseInt(packet.slot) - 5;
-		if (slot < 4) slot = 3 - slot;
-		if (packet.windowId != 0) slot = Number.parseInt(packet.slot);
+		if (slot < 4) {slot = 3 - slot;}
+		if (packet.windowId != 0) {slot = Number.parseInt(packet.slot);}
```
Comment (DataM0del) [inline index.js:1028]:
```suggestion
		if (slot < 4) slot = 3 - slot;
		if (packet.windowId != 0) slot = Number.parseInt(packet.slot);
```

**Review by DataM0del — COMMENTED** [review]
a bunch of noise in the diff from renaming, formatting, and adding brackets to things unrelated to the PR. Please apply the suggested changes to remove the noise.

**Comment by Roblox-Thot** [comment]
> Would this be possible to do with the Shop NPCs in EggWars?

Haven't looked into the eggwars since I haven't played it yet. but i don't see any reason looking for the packet wouldn't work.

**Review by DataM0del — APPROVED** [review]

**Comment by DataM0del** [comment]
it's basically the same system as this one except with a NPC and there's swords and stuff, it just calls out to /buy anyways so you just do that whenever they click on an item in the window.

**Comment by DataM0del** [comment]
the first line of the shop items (just so you can CTRL + F it in DevTools):
```js
Shop.addItem("Blocks", "white_wool", 4, "iron_ingot", 16, "Wool");
```

**Comment by Roblox-Thot** [comment]
I'm aware, I just don't see the Store NPCs (yet?) so I'll need to work on fixing that before working on a custom UI for the shop, unless i just made a /shop command.

**Comment by DataM0del** [comment]
they're probably an entity, but idk

**Comment by DataM0del** [comment]
image
Apparently this won't be merged.

**Comment by joeengo** [comment]
> image image @Roblox-Thot Apparently this won't be merged.

who the fuck is this guy

**Comment by DataM0del** [comment]
> > image image @Roblox-Thot Apparently this won't be merged.
> 
> who the fuck is this guy

what guy?

**Comment by 7GrandDadPGN** [comment]
man yall are babies, Issue closed.

**Comment by DataM0del** [comment]
> man yall are babies, Issue closed.

erm :nerd_face: :point_up::point_up: it's actually a PR, or Pull Request, not an Issue.

**Comment by joeengo** [comment]
> > > image image @Roblox-Thot Apparently this won't be merged.
> > 
> > 
> > who the fuck is this guy
> 
> what guy?

you are the fucking guy
who are you

**Comment by DataM0del** [comment]
> > > > image image @Roblox-Thot Apparently this won't be merged.
> > > 
> > > 
> > > who the fuck is this guy
> > 
> > 
> > what guy?
> 
> you are the fucking guy who are you

no name because he no got no 5 billion subscribers and 5 billion views and no flexing his non-existent AR2 disabler bypass in a video with 9999999 views.
I seriously do not care about your no name crap because we're all no names to someone, just ask the guy scrolling through brainrot shorts who you are and they won't know, they'll only know the people showing up in their doom scrolling session.

**Comment by joeengo** [comment]
> > > > > image image @Roblox-Thot Apparently this won't be merged.
> > > > 
> > > > 
> > > > who the fuck is this guy
> > > 
> > > 
> > > what guy?
> > 
> > 
> > you are the fucking guy who are you
> 
> no name because he no got no 5 billion subscribers and 5 billion views and no flexing his non-existent AR2 disabler bypass in a video with 9999999 views. I seriously do not care about your no name crap because we're all no names to someone, just ask the guy scrolling through brainrot shorts who you are and they won't know, they'll only know the people showing up in their doom scrolling session.

cope
